libcnb/src/layer/handling.rs

```diff
@@ -233,27 +233,61 @@ pub enum ReadLayerError {
 }
 
 #[derive(thiserror::Error, Debug)]
-pub enum WriteLayerError {
+pub enum WriteLayerMetadataError {
     #[error("Unexpected I/O error while writing layer metadata: {0}")]
     IoError(#[from] std::io::Error),
 
     #[error("Error while writing layer content metadata TOML: {0}")]
     TomlFileError(#[from] TomlFileError),
+}
+
+#[derive(thiserror::Error, Debug)]
+#[allow(clippy::enum_variant_names)]
+pub enum WriteLayerError {
```

Comment:
(optional)
Should `WriteLayerError` be listed above the other enums here, given it's the topmost enum in the hierarchy?

Reply:
I don't have a strong opinion about the location. For consistency, putting it with the other sub-errors seems good though.

Reply:
Also no strong opinion here, I reordered them in c12b1b0fdc2a82368b1a97fe0020b3e75a4f5ec0